openasip/src/base/osal/InstructionExecutor.cc

```diff
@@ -0,0 +1,287 @@
+#include <string>
+#include <vector>
+
+#include "OperationContext.hh"
+#include "OperationPool.hh"
+#include "Operation.hh"
+#include "InstructionFormat.hh"
+
+#include "RISCVTools.hh"
+#include "Machine.hh"
+#include "BEMGenerator.hh"
+#include "BinaryEncoding.hh"
+#include "RISCVFields.hh"
+
+/**
+ * Helper function for the extern C functions.
```

Comment:
Check the doxygen style of older files such as Machine.h or Program.hh. I think the more important sentence is now last while the fact that it's meant for calling from C is secondary. Add an empty line between the brief and extended comments.